client/src/pages/dashboard/MyCourses.js

```diff
@@ -0,0 +1,197 @@
+    button: {
+        color: '#FFF',
+        background: theme.palette.primary.gradient,
+        marginTop: theme.spacing(4),
+        height: '3rem',
+    },
+}));
+
+const mockMyCourses = [
+    {
+        id: 2,
+        name: 'Art History and Visual Culture',
+    },
+    {
+        id: 3,
+        name: 'Environment and Health',
+    },
+    {
+        id: 4,
+        name: 'Neuroscience',
+    },
+];
+
+const MyCourses = () => {
+    const classes = useStyles();
+    const [school] = useState('University of Toronto');
+    const [course, setCourse] = useState('');
+    const [schoolCourses] = useState(courses);
+    const [myCourses, setMyCourses] = useState(mockMyCourses);
+
+    const addCourse = () => {
+        if (myCourses.some(c => c.id === course)) return;
+        setMyCourses([...myCourses, schoolCourses.find(c => c.id === course)]);
+        setCourse('');
+    };
+
+    const removeCourse = id => {
+        setMyCourses(myCourses.filter(c => c.id !== id));
+    };
+
+    const hanldeCourseUpdate = () => {
+        const courses = [...myCourses].map(c => c.id);
+        // axios.post('/', {courses});
+    };
+
+    return (
+        <Grid container>
+            <Grid item md={3} style={{ height: '100vh', backgroundColor: '#f1f1f1' }}>
+                Sidebar
+            </Grid>
+            <Grid item container md={9} className={classes.container}>
+                <Typography variant="h4" className={classes.university}>
+                    {school}
+                </Typography>
+                <Typography variant="h5" className={classes.title}>
+                    Your courses
+                </Typography>
+                {myCourses.length < 1 && <Typography>No course selected.</Typography>}
+                {myCourses.map(c => {
+                    if (c.id === course) {
+                        return (
+                            <ListItem key={c.id} className={classes.list_item}>
+                                <Typography>{c.name}</Typography>
+                                <div className={classes.btn_group}>
+                                    <IconButton
+                                        edge="end"
+                                        onClick={() => removeCourse(c.id)}
+                                    >
+                                        <DeleteIcon />
+                                    </IconButton>
+                                    <IconButton edge="end" onClick={() => setCourse('')}>
+                                        <HighlightOffIcon />
+                                    </IconButton>
+                                </div>
+                            </ListItem>
+                        );
+                    }
+                    return (
+                        <ListItem key={c.id} className={classes.list_item}>
+                            <Typography>{c.name}</Typography>
+                            <IconButton edge="end" onClick={() => setCourse(c.id)}>
```

Comment:
Because selectedId is used for both selecting a course to be removed and selecting a course to add it, i think it will show up in the menu as selected when you click on the edit sign.

Reply:
I have changed it to separate states for selecting and deleting courses.